### PR TITLE
Datetime picker

### PR DESCRIPTION
**Pull request by spicalous**

### CODE REVIEW

File: test/unit/component/date-time-input/date-time-input-test.js [hunk 1]
```diff
@@ -17,18 +17,40 @@ describe('Component | Date Time Input', function () {
...
+        TestUtils.Simulate.click(TestUtils.findRenderedDOMComponentWithClass(dateTimeInput, 'date-time-input__button'));
+      });
+
+      afterEach(() => {
+        dateTimeInput = null;
       });
-    });
 
+      it('shows the date picker when focused', (done) => {
+        then(() => {
```
Comment (HioLeong) [inline test/unit/component/date-time-input/date-time-input-test.js]: Chai as a promise might make this more readable

File: app/component/date-time-input/date-time-input.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    // we then need to check if the user intended to submit
+    if (ReactDOM.findDOMNode(this).contains(event.target)) { // eslint-disable-line react/no-find-dom-node
+      this.props.onSubmit(event);
+    }
+    this.setState({ focused: false });
+  },
+
+  render() {
+    const button = this.state.focused ?
+      <div className="date-time-input__button" onClick={this.submitEvent}>Submit</div> :
```
Comment (HioLeong) [inline app/component/date-time-input/date-time-input.js]: Refactoring this? Seems like a duplicate for the html?

Comment (spicalous) [inline app/component/date-time-input/date-time-input.js]: Original
```
const button = this.state.focused ?
   <div className="date-time-input__button" onClick={this.submitEvent}>Submit</div> :
   <div className="date-time-input__button" onClick={this.toggleFocus}>ClockIcon</div>;

return (
   ...
   {button}
   ...
);
```

or 

Alternative
```
const buttonContent = this.state.focused ? 'Submit' : 'ClockIcon';
const buttonOnClick = this.state.focused ? this.submitEvent : this.toggleFocus;

return (
   ...
   <div className="date-time-input__button" onClick={buttonOnClick}>buttonContent</div>
   ...
);
```

Or any other suggestion?

File: test/unit/component/date-time-input/date-time-input-test.js (hunk 1 above)
Comment (spicalous) [inline test/unit/component/date-time-input/date-time-input-test.js]: So I looked at chai-as-promised and its not what we want

The problem with `Simulate.click` is that I cant assert the DOM element in the same event loop

http://brandonokert.com/2015/08/04/TestingInReact/#You_Cannot_Check_State/Render_or_Simulate_Right_After_Simulating_an_Event

"When we clicked the account, we triggered a setState, which will in turn change the actual state, as well as re-render. However, both of these are not done synchronously - they are added to the next slot in the event loop."

So im actually **creating** the async event (then) to assert the DOM element in the next event loop

chai as promised is for testing async functions which we do not have

File: app/component/date-time-input/date-time-input.js (hunk 2 above)
Comment (HioLeong) [inline app/component/date-time-input/date-time-input.js]: Alternative is good, as long as we just maintain one html structure.

Comment (HioLeong) [inline app/component/date-time-input/date-time-input.js]: http://fontawesome.io/icon/clock-o/

Also, is this something we want?

File: app/component/date-time-input/month-grid.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  getInitialState() {
+    return {};
+  },
+
+  onDayClicked(day) {
+    this.props.onChange(day);
+  },
+
+  render() {
+    if (this.props.monthToDisplay) {
```
Comment (HioLeong) [inline app/component/date-time-input/month-grid.js]: Damn this is one big piece of render

**Pull request merged**